src/components/views/rooms/SendMessageComposer.js

```diff
@@ -445,13 +445,11 @@ export default class SendMessageComposer extends React.Component {
 
     _onPaste = (event) => {
         const {clipboardData} = event;
-        // Prioritize text on the clipboard over files as Office on macOS puts a bitmap
-        // in the clipboard as well as the content being copied.
-        if (clipboardData.files.length && !clipboardData.types.some(t => t === "text/plain")) {
-            // This actually not so much for 'files' as such (at time of writing
-            // neither chrome nor firefox let you paste a plain file copied
-            // from Finder) but more images copied from a different website
-            // / word processor etc.
+        // Prioritize text on the clipboard over files if RTF is present as Office on macOS puts a bitmap
+        // in the clipboard as well as the content being copied. Modern versions of Office seem to not do this anymore.
+        // We check text/rtf instead of text/plain as when copy+pasting a file from Finder it puts the filename
+        // in as text/plain which we want to ignore.
+        if (clipboardData.files.length && !clipboardData.types.includes("text/rtf")) {
```

Comment:
Surely we should still be falling back to text/plain too?

Reply:
as the comment says, this heuristic is based on rtf because Finder puts both the File and `test/plain` with the filename in it. Office puts text/rtf text/html text/plain and in some prior version also a bitmap

Reply:
but we're not dealing with just Apple here, this is also a thing for Windows and Linux, though the Office issue in particular is actually documented.

Reply:
The behaviour should be to use the file 1st except in the case of Office/other known faulty things to enable copy pasting files where an annotation of the file name is also attached.

Reply:
yes, and the point I'm trying to make is that not all of those things publish `text/rtf`.

Reply:
Can you give an example of one so I can see how else to work around it? Do you have any suggestions on how to work around them whilst fixing the regression of breaking copy paste from Finder?

Reply:
Based on your original report `text/rtf` was present

https://github.com/vector-im/element-web/issues/13985#issue-636493170